### DISCUSSION TITLE
Display stopped working after using PROGMEM sketch

### DISCUSSION THREAD

**Discussion by Wanhallen** [moononournation/Arduino_GFX#206] [Q&A]
Hello. Im currently using Arduino_GFX with my gc9a01 round display. It worked fine for me, until i used this particular sketch : ImgViewerPROGMEM.

The problem is: my LCD stopped working after this one code. Idk why. It worked for once. After arduino reboot, screen just went black and nothing happens. I've tried different : arduino nano, gc9a01 screens, wires, sketches ofc and even different PC. Nothing.

The funny thing is: i've also tried a st7789 display with arduino_gfx libs, Everything was ok untill this exact ImgViewerPROGMEM sketch )

It may be just a bad coincidence, but i just wanna make sure. Can this issue caused by a code? May be something with PROGMEM thing?
Thanks for answers )

**Comment by moononournation** [comment]
After compiled, it will show the flash and memory usage. Can you show this information here?

**Reply by Wanhallen** [reply]
Soryy, used google translate. Hope u understand.

Sketch uses 26554 bytes (86%) of device memory. A total of 30720 bytes are available.
Global variables use 436 bytes (21%) of dynamic memory, leaving 1612 bytes for lunch. Maximum: 2048 bytes.

**Reply by moononournation** [reply]
https://user-images.githubusercontent.com/13228891/191761874-9b213b58-34eb-4640-b42e-810fe35ce332.MOV

**Reply by moononournation** [reply]
Just tested it is ok with latest software and ILI9341.

**Reply by Wanhallen** [reply]
Can u please unplug all wires. Use another sketch and then connect everything once again (with progmem sketch). This how i get my problem.

**Reply by moononournation** [reply]
This dev board uploaded PDQgraphicstest sketch originally, then upload ImgViewerPROGMEM sketch without problems.

**Reply by Wanhallen** [reply]
Got it. Ok then, i guess im gonna try ILI9341 screen. But this is still a mistery to me. Im newbee at all this arduino stuff, yet you lib worked just fine. But after a while its just stopped. =\
Also, if anyone of you, guys, can test this with gc9a01 or st7789 please share your results.

**Reply by moononournation** [reply]
Do you have another dev board? e.g. ESP32?

**Reply by moononournation** [reply]
If you would to display graphics, Arduino Nano have many limitations.

**Reply by laoadam** [reply]
Hi,
I just can't make ESP32+GC9A01 works at GFX for Arduino, the same sketch works with ILI9341，not with GC9A01 with the: Arduino_GFX *gfx = new Arduino_GC9A01(bus, 7 /* RST */, 0 /* rotation */, true /* IPS */); changes.
any thing trick ?

**Reply by moononournation** [reply]
try the built-in example, follow the pin map, without any code changes first

**Reply by laoadam** [reply]
Thank you.
I tested the example from GFX for Arduino, only changes is the gfx=...

**Reply by moononournation** [reply]
any connection photo?

**Reply by laoadam** [reply]
yes, but the same wiring tested by example of TFT_eSPI at the same configuration works well. the TFT_eSPI test is for make sure the wiring only. 
TFT_eSPI: 
#define TFT_MOSI 23
#define TFT_SCLK 18
#define TFT_CS     5  // Chip select control pin
#define TFT_DC     27   // Data Command control pin
#define TFT_RST    33  // Reset pin (could connect to RST pin)

**Reply by moononournation** [reply]
Have you mixed up 2 libraries? This is Arduino_GFX.

**Reply by laoadam** [reply]
Nop. I used example of TFT--- to test the wiring only. then upload the example from "GFX_Library_for_Arduino". actually none of the example works GC9A01 but all work for ILI9341.

**Reply by moononournation** [reply]
ok, then any GC9A01 connection photo?

**Reply by laoadam** [reply]
Yes. I didn't find the upload image here, the link is a hosting. The picture taken with a TFT clock example.
https://ibb.co/VqZn7DL
Thanks.

**Reply by moononournation** [reply]
https://user-images.githubusercontent.com/13228891/193070625-023a30d1-ddc1-43f3-bf13-1bf84cc7db55.MOV
just make a quick test, everything works fine.

**Reply by laoadam** [reply]
I selected board: ESP32-WROOM-DA Module, is it OK? and got rebooting, what can be?

if add: Arduino_GFX *gfx = new Arduino_GC9A01(bus, DF_GFX_RST, 7 /* RST */, 0 /* rotation */, true /* IPS */);
got a full screen snow; without 'DF_GFX_RST, ' , got keep rebooting. 
https://ibb.co/5MvVBRs

are there any special setting?

serial print:
16:09:12.677 -> ets Jun  8 2016 00:22:57
16:09:12.711 -> 
16:09:12.711 -> rst:0x8 (TG1WDT_SYS_RESET),boot:0x13 (SPI_FAST_FLASH_BOOT)
16:09:12.711 -> configsip: 0, SPIWP:0xee
16:09:12.711 -> clk_drv:0x00,q_drv:0x00,d_drv:0x00,cs0_drv:0x00,hd_drv:0x00,wp_drv:0x00
16:09:12.711 -> mode:DIO, clock div:1
16:09:12.711 -> load:0x3fff0030,len:1344
16:09:12.711 -> load:0x40078000,len:13864
16:09:12.711 -> load:0x40080400,len:3608
16:09:12.711 -> entry 0x400805f0
